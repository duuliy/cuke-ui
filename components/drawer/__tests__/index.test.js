import assert from "power-assert";
import React from "react";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Drawer from "../index";
import Button from "../../button";

describe("<Drawer/>", () => {
  it("should render a <Drawer/> components", () => {
    const wrapper = render(
      <div>
        <Drawer title="基本使用" visible>
          <span>基本使用</span>
        </Drawer>
        <Drawer title="基本使用" visible width={500}>
          <span>无 footer</span>
        </Drawer>
        <Drawer title="基本使用" visible width={500} zIndex={9999}>
          <span>无 footer</span>
        </Drawer>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should can not clicked with set maskClosable false", () => {
    const onCancelClick = jest.fn();
    const wrapper = shallow(
      <Drawer visible={true} maskClosable={false} onCancel={onCancelClick}>
        <span>关闭回调</span>
      </Drawer>
    );
    wrapper.find(".cuke-drawer").simulate("click");
    expect(onCancelClick).not.toHaveBeenCalled();
  });

  it("should render custom direction", () => {
    const wrapper = render(
      <div>
        <Drawer visible placement="left">
          <span>关闭回调</span>
        </Drawer>
        <Drawer visible placement="top">
          <span>关闭回调</span>
        </Drawer>
        <Drawer visible placement="right">
          <span>关闭回调</span>
        </Drawer>
        <Drawer visible placement="bottom">
          <span>关闭回调</span>
        </Drawer>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should can trigger onClose event", () => {
    const onCancelClick = jest.fn();
    const wrapper = shallow(
      <Drawer title="关闭回调" visible onClose={onCancelClick}>
        <span>关闭回调</span>
      </Drawer>
    );
    wrapper.find(".cuke-drawer-close").simulate("click");
    expect(onCancelClick).toHaveBeenCalled();
  });

  it("should render custom footer", () => {
    const wrapper = shallow(
      <Drawer title="关闭回调" visible footer={<Button>11</Button>}>
        <span>关闭回调</span>
      </Drawer>
    );
    assert(wrapper.find(".cuke-drawer-footer").length === 1);
    assert(wrapper.find(Button).length === 1);
  });

  it("should can not render title", () => {
    const wrapper = shallow(
      <Drawer visible>
        <span>关闭回调</span>
      </Drawer>
    );
    assert(wrapper.find(".cuke-drawer-no-title").length === 1);
  });

  it("should can not find cuke-drawer-no-title when title exist", () => {
    const wrapper = shallow(
      <Drawer visible title="test">
        <span>关闭回调</span>
      </Drawer>
    );
    assert(wrapper.find(".cuke-drawer-no-title").length === 0);
  });

  it("should can not render custom footer", () => {
    const wrapper = shallow(
      <Drawer title="关闭回调" visible footer={null}>
        <span>关闭回调</span>
      </Drawer>
    );
    assert(wrapper.find(".cuke-drawer-footer").length === 0);
  });

  it("should disabled scroll when did update", () => {
    const wrapper = shallow(
      <Drawer title="关闭回调" visible>
        <span>关闭回调</span>
      </Drawer>
    );
    wrapper.update();
    assert(document.body.style.overflow === "");
  });
  it("should enable scroll and disabled scroll", () => {
    const wrapper = shallow(
      <Drawer visible={false}>
        <span>1</span>
      </Drawer>
    );
    wrapper.update();
    expect(document.body.style.paddingRight).toBe("");
    wrapper.setState({
      visible: true
    });
    wrapper.update();
    setTimeout(() => {
      expect(document.body.style.paddingRight).toBe("15px");
    });
  });
  it("should set body style when did update", () => {
    const wrapper = shallow(
      <Drawer visible={false}>
        <span>1</span>
      </Drawer>
    );
    wrapper.setProps({ visible: true });
    wrapper.update();
    expect(document.body.style.overflow).toEqual("hidden");
    expect(document.body.style.paddingRight).toEqual("15px");
    wrapper.setProps({ visible: false });
    wrapper.update();
    expect(document.body.style.overflow).toEqual("");
    expect(document.body.style.paddingRight).toEqual("0px");
  });

  it("should find custom wrapperClassName ", () => {
    const wrapper = shallow(
      <Drawer wrapperClassName="test" visible>
        1111
      </Drawer>
    );
    assert(wrapper.find(".test").length === 1);
  });
});
