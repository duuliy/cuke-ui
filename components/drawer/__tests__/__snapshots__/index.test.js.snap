// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer/> should render a <Drawer/> components 1`] = `
<div>
  <div
    class="cuke-drawer-mask cuke-drawer-mask-show"
  />
  <div
    class="cuke-drawer-wrap cuke-drawer-wrap-visible"
    tabindex="-1"
  >
    <div
      class="cuke-drawer cuke-drawer-open cuke-drawer-right"
      style="width:300px;height:100%;z-index:999"
    >
      <section
        class="cuke-drawer-header"
      >
        <h2
          class="cuke-drawer-title"
        >
          基本使用
        </h2>
        <svg
          class="cuke-drawer-close"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </section>
      <section
        class="cuke-drawer-content"
      >
        <span>
          基本使用
        </span>
      </section>
    </div>
  </div>
  <div
    class="cuke-drawer-mask cuke-drawer-mask-show"
  />
  <div
    class="cuke-drawer-wrap cuke-drawer-wrap-visible"
    tabindex="-1"
  >
    <div
      class="cuke-drawer cuke-drawer-open cuke-drawer-right"
      style="width:500px;height:100%;z-index:999"
    >
      <section
        class="cuke-drawer-header"
      >
        <h2
          class="cuke-drawer-title"
        >
          基本使用
        </h2>
        <svg
          class="cuke-drawer-close"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </section>
      <section
        class="cuke-drawer-content"
      >
        <span>
          无 footer
        </span>
      </section>
    </div>
  </div>
  <div
    class="cuke-drawer-mask cuke-drawer-mask-show"
  />
  <div
    class="cuke-drawer-wrap cuke-drawer-wrap-visible"
    tabindex="-1"
  >
    <div
      class="cuke-drawer cuke-drawer-open cuke-drawer-right"
      style="width:500px;height:100%;z-index:9999"
    >
      <section
        class="cuke-drawer-header"
      >
        <h2
          class="cuke-drawer-title"
        >
          基本使用
        </h2>
        <svg
          class="cuke-drawer-close"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </section>
      <section
        class="cuke-drawer-content"
      >
        <span>
          无 footer
        </span>
      </section>
    </div>
  </div>
</div>
`;

exports[`<Drawer/> should render custom direction 1`] = `
<div>
  <div
    class="cuke-drawer-mask cuke-drawer-mask-show"
  />
  <div
    class="cuke-drawer-wrap cuke-drawer-wrap-visible"
    tabindex="-1"
  >
    <div
      class="cuke-drawer cuke-drawer-open cuke-drawer-no-title cuke-drawer-left"
      style="width:300px;height:100%;z-index:999"
    >
      <section
        class="cuke-drawer-header"
      >
        <h2
          class="cuke-drawer-title"
        />
        <svg
          class="cuke-drawer-close"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </section>
      <section
        class="cuke-drawer-content"
      >
        <span>
          关闭回调
        </span>
      </section>
    </div>
  </div>
  <div
    class="cuke-drawer-mask cuke-drawer-mask-show"
  />
  <div
    class="cuke-drawer-wrap cuke-drawer-wrap-visible"
    tabindex="-1"
  >
    <div
      class="cuke-drawer cuke-drawer-open cuke-drawer-no-title cuke-drawer-top"
      style="width:300px;height:300px;z-index:999"
    >
      <section
        class="cuke-drawer-header"
      >
        <h2
          class="cuke-drawer-title"
        />
        <svg
          class="cuke-drawer-close"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </section>
      <section
        class="cuke-drawer-content"
      >
        <span>
          关闭回调
        </span>
      </section>
    </div>
  </div>
  <div
    class="cuke-drawer-mask cuke-drawer-mask-show"
  />
  <div
    class="cuke-drawer-wrap cuke-drawer-wrap-visible"
    tabindex="-1"
  >
    <div
      class="cuke-drawer cuke-drawer-open cuke-drawer-no-title cuke-drawer-right"
      style="width:300px;height:100%;z-index:999"
    >
      <section
        class="cuke-drawer-header"
      >
        <h2
          class="cuke-drawer-title"
        />
        <svg
          class="cuke-drawer-close"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </section>
      <section
        class="cuke-drawer-content"
      >
        <span>
          关闭回调
        </span>
      </section>
    </div>
  </div>
  <div
    class="cuke-drawer-mask cuke-drawer-mask-show"
  />
  <div
    class="cuke-drawer-wrap cuke-drawer-wrap-visible"
    tabindex="-1"
  >
    <div
      class="cuke-drawer cuke-drawer-open cuke-drawer-no-title cuke-drawer-bottom"
      style="width:300px;height:300px;z-index:999"
    >
      <section
        class="cuke-drawer-header"
      >
        <h2
          class="cuke-drawer-title"
        />
        <svg
          class="cuke-drawer-close"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </section>
      <section
        class="cuke-drawer-content"
      >
        <span>
          关闭回调
        </span>
      </section>
    </div>
  </div>
</div>
`;
